### PR TITLE
[TASK] Revise "Accessing site configuration" chapter

### PR DESCRIPTION
**Pull request by brotkrueml**
- Enhance content with corresponding links
- Adjust formatting

Releases: main, 11.5

### CODE REVIEW

**Comment by brotkrueml** [comment]
@linawolf I wanted to add the SiteFinder API with a code snippet, but the snippet generator does not add the "getSiteByRootPageId" to the class methods. Maybe you can have a look into it?

**Comment by linawolf** [comment]
Did you make a `composer update` so you have the current main version of the Core installed? And where is your codesnippet configuration? Did you forget to attach it?

**Comment by brotkrueml** [comment]
> @brotkrueml Did you make a `composer update` so you have the current main version of the Core installed? And where is your codesnippet configuration? Did you forget to attach it?

Nope, but that does not matter, as the SiteFinder with all four public methods is available since v9. I omitted it and just preserved the current manual list of methods. From my side. it is ready. The automatic snippet may be nice to have of this PR but it is not mandatory. My focus was on the tabs with Composer/legacy.

**Comment by linawolf** [comment]
If I should have a look in why your configuration does not work it would be good to have your configuration. Otherwise i can review like it is

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 1]
```diff
@@ -1,108 +1,121 @@
...
-Let's look at both cases in detail:
 
 Accessing the current site object
 =================================
 
-When rendering the frontend or backend, TYPO3 builds an HTTP request object
-through a `PSR-15` middleware stack and enriches that with information.
-Part of that information are the :php:`Site` and :php:`SiteLanguage` objects. Both
-objects are available as attributes on the current request object.
+When rendering frontend or backend, TYPO3 builds an HTTP request object through
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:27]:
```suggestion
When rendering the frontend or backend, TYPO3 builds an HTTP request object through
```

**Comment by brotkrueml** [comment]
> If I should have a look in why your configuration does not work it would be good to have your configuration. Otherwise i can review like it is

Will open an issue in the snippet repository for it. Sorry for the buzz, you can review the PR as is.

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 2]
```diff
@@ -1,108 +1,121 @@
...
+a :ref:`PSR-15 middleware stack <request-handling>` and enriches it with
+information. Part of that information are the
+:php:`\TYPO3\CMS\Core\Site\Entity\Site` and
+:php:`\TYPO3\CMS\Core\Site\Entity\SiteLanguage` objects. Both objects are
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:31]:
```suggestion
information. Part of that information are the objects
:php:`\TYPO3\CMS\Core\Site\Entity\Site` and
:php:`\TYPO3\CMS\Core\Site\Entity\SiteLanguage`. Both objects are
```

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 3]
```diff
@@ -1,108 +1,121 @@
...
+available as :ref:`attributes <typo3-request-attributes>` in the current
+:ref:`request object <typo3-request>`.
 
 Depending on the context, there are two main ways to access them:
 
-* via the `PSR-7` HTTP ServerRequest object directly - for example in a PSR-15 middleware
-  or the admin panel
-* via :php:`$GLOBALS['TYPO3_REQUEST']` - everywhere you don't have a ServerRequest object
+-   via the PSR-7 HTTP request object directly - for example in a PSR-15
+    middleware or the :doc:`admin panel <ext_adminpanel:Index>`
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:38]:
```suggestion
   or the :doc:`admin panel <ext_adminpanel:Index>`
```
Why the admin panel? Aren't we talking about using the site Configuration in frontend and backend rendering here?

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 4]
```diff
@@ -1,108 +1,121 @@
...
+-   via :php:`$GLOBALS['TYPO3_REQUEST']` - everywhere you do not have a
+    request object
 
 ..  hint::
-    The first method is preferred if possible as :php:`$GLOBALS['TYPO3_REQUEST']` was
-    deprecated in 9.2 and will be removed in future versions.
+    The first method is preferred if possible as :php:`$GLOBALS['TYPO3_REQUEST']`
+    was deprecated in 9.2 and will be removed in a future version.
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:44]:
```suggestion
    was deprecated in TYPO3 v9.2 and will be removed in a future version.
```

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 5]
```diff
@@ -1,108 +1,121 @@
...
+
 .. index:: pair: Site handling; SiteFinder
 
 Finding a site object
 =====================
 
-When you need to access site configuration for a specific page ID or by identifier,
-you can use the :php:`SiteFinder` (:php:`\TYPO3\CMS\Core\Site\SiteFinder`).
+When you need to access the site configuration for a specific page ID or by an
+identifier, you can use the :php:`\TYPO3\CMS\Core\Site\SiteFinder` class.
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:70]:
```suggestion
When you need to access the site configuration for a specific page ID or by a
site identifier, you can use the class :php:`\TYPO3\CMS\Core\Site\SiteFinder`.
```

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 6]
```diff
@@ -1,108 +1,121 @@
...
+All methods for finding a specific site throw an exception, if no site was found.
 
 
 .. index:: pair: Site handling; Site object
 
-The Site object
+The site object
 ===============
 
 Now we know how to find a :php:`Site` object, but what can it do?
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:90]: And what if you don't know? I would not assume this

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst [hunk 7]
```diff
@@ -1,108 +1,121 @@
...
 First of all, it gives us access to the site configuration options via
 
-- :php:`getConfiguration()`: returns the complete configuration
-- :php:`getAttribute()`: returns a specific configuration attribute (root level configuration only)
+-   :php:`getConfiguration()`: returns the complete configuration
+-   :php:`getAttribute()`: returns a specific configuration attribute (root
+    level configuration only)
 
-It additionally provides methods for accessing related objects (languages / errorHandling):
+Additionally, it provides methods for accessing related objects (languages /
```
Comment (linawolf) [inline Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst:98]:
```suggestion
Additionally, the :php:`Site` provides methods for accessing related objects (languages /
```
I would repeat this here, it was two paragraphs up we mentioned what we talk about

File: Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst (hunk 3 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/SiteHandling/AccessingSiteConfiguration.rst]: Good objection - adjusted the examples.

**Comment by brotkrueml** [comment]
> > If I should have a look in why your configuration does not work it would be good to have your configuration. Otherwise i can review like it is

Aaargh, one method is marked as internal. Looked at it multiple times and haven't realized it. Need more rests ...

**Comment by linawolf** [comment]
> Aaargh, one method is marked as internal. Looked at it multiple times and haven't realized it. Need more rests ...

We might need better error messages then

**Review by linawolf — APPROVED** [review]

**Pull request merged**